File: src/com/rian/difficultycalculator/beatmap/constants/SampleBank.java
package com.rian.difficultycalculator.beatmap.constants;

/**
 * Represents available sample banks.
 */
public enum SampleBank {
    none,
    normal,
    soft,
    drum;

    /**
     * Converts an integer value to its sample bank counterpart.
     *
     * @param value The value to convert.
     * @return The sample bank counterpart of the given value.
     */
    public static SampleBank parse(int value) {
        switch (value) {
            case 1:
                return normal;
            case 2:
                return soft;
            case 3:
                return drum;
            default:
                return none;
        }
    }

    /**
     * Converts a string value to its sample bank counterpart.
     *
     * @param value The value to convert.
     * @return The sample bank counterpart of the given value.
     */
    public static SampleBank parse(String value) {
        switch (value) {
            case "Normal":
                return normal;
            case "Soft":
                return soft;
            case "Drum":
                return drum;
            default:
                return none;
        }
    }
}

File: src/com/rian/osu/beatmap/constants/SampleBank.kt
package com.rian.osu.beatmap.constants

/**
 * Represents available sample banks.
 */
enum class SampleBank(
    /**
     * The prefix of audio files representing this sample bank.
     */
    @JvmField
    val prefix: String
) {
    None(""),
    Normal("normal"),
    Soft("soft"),
    Drum("drum");

    companion object {
        /**
         * Converts an integer value to its sample bank counterpart.
         *
         * @param value The value to convert.
         * @return The sample bank counterpart of the given value.
         */
        @JvmStatic
        fun parse(value: Int) =
            when (value) {
                1 -> Normal
                2 -> Soft
                3 -> Drum
                else -> None
            }

        /**
         * Converts a string value to its sample bank counterpart.
         *
         * @param value The value to convert.
         * @return The sample bank counterpart of the given value.
         */
        @JvmStatic
        fun parse(value: String?) =
            when (value) {
                "Normal" -> Normal
                "Soft" -> Soft
                "Drum" -> Drum
                else -> None
            }
    }
}
